packages/@react-stately/selection/src/SelectionManager.ts

```diff
@@ -480,7 +480,7 @@ export class SelectionManager implements MultipleSelectionManager {
     }
 
     let item = this.collection.getItem(key);
-    if (!item || (item.type === 'cell' && !this.allowsCellSelection)) {
+    if (!item || (item.type === 'cell' && !this.allowsCellSelection) || (item.type === 'item' && item?.props?.isTrigger)) {
```

Comment:
What is this for? Would be good to avoid menu-specific code in here

Reply:
I needed a way to make SubMenuTriggers non-selectable but NOT disabled since they should be keyboard navigable. Without this code, if a user specified the key of a submenu trigger menu item in their "selectedKeys" array, it would be considered selected from the SelectionManager's point of view

Reply:
What do you think about a different prop name for this? Maybe something more generic like `props?.selectionDisabled`?

Reply:
can that logic be in `useMenuItem`? seems like programmer error to mark a trigger as selected, but we could just not display it as selected at least

Reply:
off the top of my head, I think we'd have to add a prop or something to useSelectableItem since it currently uses `manager.canSelectItem` as its check for `allowsSelection`. I'll dig

Reply:
One alternative is that I don't use the item props (bar the `onFocus` to maintain focusedKey tracking when a user presses on a trigger in a submenu Tray)  from `useSelectableItem` in `useMenuItem` if it is a submenu trigger, but it still exposes the submenu trigger item as a valid selectable key from the SelectionManager's point of view which feels risky since we have various checks all over the place that use `manager.isSelected(key)`.